### PR TITLE
Feature/queue framework

### PR DESCRIPTION
**Pull request by ghbm-itk**
Linear and queue framework merged into one project with two separate framework files.

Get constants.py removed in favor of config.py

### CODE REVIEW

**Comment by ghbm-itk** [comment]
Should the framework raise and unchaught error if the max retry count is reached?
I just tested a queue robot that failed at startup, but then just kept restarting because the queue wasn't empty

File: README.md [hunk 1]
```diff
@@ -1,32 +1,61 @@
-# Robot-Framework V1
+# Robot-Framework V2
 
 This repo is meant to be used as a template for robots made for [OpenOrchestrator](https://github.com/itk-dev-rpa/OpenOrchestrator).
 
 ## Quick start
 
-1. To use the template simply use the repo as a template as shown [here](https://docs.github.com/en/repositories/creating-and-managing-repositories/creating-a-repository-from-a-template).
+1. To use this template simply use the repo as a template as shown [here](https://docs.github.com/en/repositories/creating-and-managing-repositories/creating-a-repository-from-a-template).
```
Comment (rimi-itk) [inline README.md:7]: You should never use “here” as a link text. Consider something like

```suggestion
1. To use this template simply use this repo as a template (cf. [Creating a repository from a template](https://docs.github.com/en/repositories/creating-and-managing-repositories/creating-a-repository-from-a-template)).
```

File: README.md [hunk 2]
```diff
@@ -1,32 +1,61 @@
...
 
 ## Requirements
 Minimum python version 3.10
 
 ## Flow
 
-The flow of the framework is sketched up in the following illustration:
+This framework contains two different flows: A linear and a queue based.
+You should only ever use one at a time. You choose which one by going into `robot_framework/__main__.py`
+and uncomment the framework you want. They are both disabled by default and an error will be
```
Comment (rimi-itk) [inline README.md:37]:
```suggestion
and uncommenting the framework you want. They are both disabled by default and an error will be
```

File: robot_framework/config.py [hunk 3]
```diff
@@ -3,7 +3,25 @@
...
+# Whether the robot should be marked as failed if MAX_RETRY_COUNT is reached.
+FAIL_ROBOT_ON_TOO_MANY_ERRORS = True
+
 # Error screenshot config
 SMTP_SERVER = "smtp.aarhuskommune.local"
 SMTP_PORT = 25
 SCREENSHOT_SENDER = "robot@friend.dk"
+
+# Constant/Credential names
+ERROR_EMAIL = "Error Email"
```
Comment (rimi-itk) [inline robot_framework/config.py:15]: Is this supposed to be an email address? Is so, state that in the documentation (the comment).

Does it make sense to let `None` be the default value and add a check in the framework for checking that a value has been set? And maybe even that the value is a valid email address?

**Review by rimi-itk — CHANGES_REQUESTED** [review]
Looks good. Some minor comments and suggestions added.

File: robot_framework/config.py (hunk 3 above)
Comment (ghbm-itk) [reply robot_framework/config.py:15]: No it's the name of the constant in OpenOrchestrator. In our case it's "Error Email".

This line is also to show a suggestion on how to define which constants the robot uses.

File: README.md (hunk 2 above)
Comment (ghbm-itk) [reply README.md:37]: I disagree

File: robot_framework/config.py (hunk 3 above)
Comment (rimi-itk) [reply robot_framework/config.py:15]: I don't understand this. Is the value, i.e. `"Error Email"`, the name of a constant that the robot uses to get a value defined elsewhere? Can I use other values than `"Error Email"`?

Comment (ghbm-itk) [reply robot_framework/config.py:15]: It's defined in OpenOrchestrator like this:

image

And used in `exceptions.py` like this:

`error_email = orchestrator_connection.get_constant(config.ERROR_EMAIL).value`

Comment (ghbm-itk) [reply robot_framework/config.py:15]: It enables us to change the value in the future without changing the source code.

**Review by rimi-itk — APPROVED** [review]

**Pull request merged**